src/containers/oracle.js

```diff
@@ -396,26 +384,57 @@ class OraclePage extends React.Component {
     }
   }
 
-  checkAllowance() {
-    try {
-      const { selectedWalletAddress } = this.props;
+  /**
+ * Send get allowance request and keep repeating itself until this.state.isApproving is false
+ * @return {[type]}
+ */
+  startCheckAllowance() {
+    console.log('startCheckAllowance', new Date());
+    const { selectedWalletAddress, onAllowance } = this.props;
+    const { oracle } = this.state;
+
+    const self = this;
 
-      this.props.onAllowance(selectedWalletAddress, this.state.oracle.topicAddress, selectedWalletAddress);
-    } catch (err) {
-      console.log(err.message);
+    // A function to repeat itself until this.state.isApproving is false
+    function startPollAllowance() {
+      if (self.state.isApproving) {
+        onAllowance(selectedWalletAddress, oracle.topicAddress, selectedWalletAddress);
+        setTimeout(startPollAllowance, ALLOWANCE_TIMER_INTERVAL);
+      }
     }
+
+    // Kick off the first round of onAllowance
+    onAllowance(selectedWalletAddress, oracle.topicAddress, selectedWalletAddress);
+    setTimeout(startPollAllowance, ALLOWANCE_TIMER_INTERVAL);
   }
 
+  /**
+   * Determine next action based on returned allowance result
+   * 1. If allowance is zero, send approve request
+   * 2. If allowance is positive but less than voteAmount, send approve(0) to reset allowance
+   * 3. If allowance is greather than or equal to voteAmount and not isApproved, do action and reset isApproving and isApproved
+   * @param  {number} allowance value
+   * @return {}
+   */
   onAllowanceReturn(allowance) {
+    if (_.isUndefined(allowance) || _.isUndefined(this.state.config)) {
```

Comment:
after testing out the flow. it looks like `allowance` doesn't get reset to `undefined` after the setResult or vote. So onAllowanceReturn() is being called over and over even after the flow is done.

This doesn't affect setResult since you can only do it once. But, this can affect vote in this case:
- Place vote to 50 BOT
- approve() for 50
- allowance, wait, allowance etc
- approve accepted for 50
- vote() called (onAllowanceReturn.allowance still returning as 50)
- user wants to vote() again for 30
- onAllowanceReturn() called when /getblockcount or /listunspent returns. (anytime componentWillReceiveProps gets called)
- will call vote() because allowance has not been reset from 50
- the 2nd vote() for 30 fails because the approve() has not been done

 Need to reset `allowanceReturn`  in `componentWillReceiveProps` after the setResult or vote is done.

Reply:
Good point. Also if allowance is not reset, if a new Vote comes in and calls onAllowance with the same value onAllowanceReturn will not be called, because componentWillReceiveProps is only triggered when there's any value change.

Reply:
Verified your theory with e2e testing. It's so slow on testnet. I plan to switch to -regtest for development testing.